### ISSUE TITLE
Parse example configuration

### ISSUE THREAD

**Issue by idefixcert** [sayotte/iscdhcp#1]
I tried to parse the example configuration from
https://github.com/42wim/isc-dhcp/blob/master/server/dhcpd.conf.example
even If I delete the comments it does not work.

**Comment by sayotte** [comment]
Can you elaborate on “does not work”?

What did you expect, and what did you see instead?

**Comment by idefixcert** [comment]
I get the error:
2019/09/26 17:27:33 iscdhcp.Decode(): error at (or just before) token "domain-name": syntax error
I expect, that I can use a isc-dhcp.conf file an parse it with the parser.

**Comment by sayotte** [comment]
Fair enough-- the parser is incomplete in its coverage, actually it only covers the ones that I needed when I created it, and I haven't needed it much since.

The `domain-name` statement is super common. Would you consider adding that support and sending me a PR? You'd need to:
* add a new type, `DomainNameOption`, similar to the one found [here](https://github.com/sayotte/iscdhcp/blob/master/statement.go#L306)
* add the "domain-name" token to the lexer, around [here](https://github.com/sayotte/iscdhcp/blob/master/lexer.go#L31)
* register the token as a reserved word in the parser, around [here](https://github.com/sayotte/iscdhcp/blob/master/parse.y#L36)
* follow the pattern from [here](https://github.com/sayotte/iscdhcp/blob/master/parse.y#L314) to make the parser emit your new type (which you defined up above); as a hint, I think it'd be something like:
    ```
    domainNameOptClause: optDomainNameTok stringConst
        {
            $$.statement = DomainNameOption{$2.str}
        };
* register the new clause you defined in the previous step as a possible sub-clause of the "option" statement, around [here](https://github.com/sayotte/iscdhcp/blob/master/parse.y#L312)

All told, that is probably ~25 lines of change, and I've just updated the project to current standards (made it a Go module, migrated to golangci-lint).

Or, I could add this myself. But my concern is that it'll actually be an onion-- after adding `domain-name` we'll see that it lacks support for another statement, then another, and so on. And, putting a fine point on it, I can't take time away from other things to dig that deep. But I _could_ probably support you with code reviews if you have the time to do the discovery/fix iterations.
